### PR TITLE
tuning of 52c0f2aaea2fa19c3ade8ed50b978acd59733d32 [internal WiFi hotspot is not working]

### PR DESCRIPTION
**Pull request by jvonau**

### CODE REVIEW

**Comment by holta** [comment]
Context:

An Android 10 phone failed to connect to both of these fresh IIAB installs: (even after they rebooted upon completion of the IIAB install)
- RPi 4 with 2GB RAM: 182-rpi4-2g-lite-BIG-RC2-0512
- RPi 3 B+: 142-rpi3bp-lite-0513

**Comment by holta** [comment]
For some reason this PR does not solve the cannot-connect-to-SSID issue on 181-rpi4-1g-lite-PR2395 = 10.8.0.30 (where it was also manually applied, i.e. to Line 29 of /lib/dhcpcd/dhcpcd-hooks/50-hostapd, and then rebooted).

Likewise this same cannot-connect-to-SSID problem appears on 182-rpi4-2g-lite-BIG-RC2-0512 = 10.8.0.22 (even after this PR was manually applied, and then rebooted).

File: roles/network/templates/hostapd/50-hostapd [hunk 1]
```diff
@@ -26,7 +26,7 @@ if [ "$interface" = "wlan0" ]; then
     fi
     # spams the logging
     #syslog info "50-iiab set ap0 spam $REASON"
-    if [ -e /sys/class/net/ap0 ] && [ "$reason" = "*CARRIER" ]; then
+    if [ -e /sys/class/net/ap0 ] && [ "$reason" = "CARRIER" ] || [ "$reason" = "NOCARRIER" ]; then
```
Comment (holta) [inline roles/network/templates/hostapd/50-hostapd]: It looks like parentheses are needed, perhaps something like the following:

```suggestion
    if [ -e /sys/class/net/ap0 ] && ( [ "$reason" = "CARRIER" ] || [ "$reason" = "NOCARRIER" ] ); then
```

Comment (jvonau) [reply roles/network/templates/hostapd/50-hostapd]: The above without () has been field tested on .30, 
>May 13 03:10:44 box.lan dhcpcd[431]: wlan0: carrier acquired
May 13 03:10:44 box.lan dhcpcd[431]: wlan0: 50-iiab CARRIER change wlan0
May 13 03:10:44 box.lan dhcpcd[431]: frequency is 6 for carrier
May 13 03:10:44 box.lan dhcpcd[431]: Using 6 for carrier
May 13 03:10:44 box.lan dhcpcd[431]: wlan0: 50-iiab set channel 6
May 13 03:10:44 box.lan dhcpcd[431]: Hostapd set for 6
May 13 03:10:44 box.lan dhcpcd[431]: wlan0: 50-iiab ap0 spam CARRIER
May 13 03:10:44 box.lan dhcpcd-run-hooks[624]: wlan0: 50-iiab ap0 spam CARRIER
May 13 03:10:49 box.lan dhcpcd[431]: wlan0: 50-iiab set ap0 up CARRIER
May 13 03:10:49 box.lan dhcpcd[431]: wlan0: 50-iiab CARRIER change wlan0
May 13 03:10:49 box.lan dhcpcd[431]: frequency is 6 for carrier
May 13 03:10:49 box.lan dhcpcd[431]: Using 6 for carrier
May 13 03:10:49 box.lan dhcpcd[431]: wlan0: 50-iiab set channel 6
May 13 03:10:49 box.lan dhcpcd-run-hooks[672]: wlan0: 50-iiab set channel 6
May 13 03:10:49 box.lan dhcpcd[431]: Hostapd set for 6
May 13 03:10:49 box.lan dhcpcd[431]: DUID 00:01:00:01:25:d8:31:74:dc:a6:32:0d:3a:a8
May 13 03:10:49 box.lan dhcpcd[431]: wlan0: IAID 32:0d:3a:a9
May 13 03:10:49 box.lan dhcpcd[431]: wlan0: adding address fe80::b66a:d4b4:a4ab:4aa6
May 13 03:10:49 box.lan dhcpcd[431]: ipv6_addaddr1: Permission denied
May 13 03:10:49 box.lan dhcpcd[431]: ipv6_addaddr1: Permission denied
May 13 03:10:49 box.lan dhcpcd[431]: wlan0: carrier lost
May 13 03:10:49 box.lan dhcpcd[431]: wlan0: 50-iiab ap0 spam NOCARRIER
May 13 03:10:54 box.lan dhcpcd[431]: wlan0: 50-iiab set ap0 up NOCARRIER

I'll tried the change on .34 works also, so it a question of esthetics.

**Comment by holta** [comment]
This PR was just tested successfully on 4 different Raspberry Pi's:
- 142-rpi3bp-lite-0513 = 10.8.0.34 (RPi 3 B+)
- 181-rpi4-1g-lite-PR239 = 10.8.0.30
- 182-rpi4-2g-lite-BIG-RC2-0512 = 10.8.0.22
- 136-rpi4-2g-lite-PR2384 = 10.8.0.62

Similar to yesterday, Line 29 of /lib/dhcpcd/dhcpcd-hooks/50-hostapd was manually modified (changing it to the following line) and then each machine rebooted &mdash; and finally each one's internal SSID was tested with a regular smartphone's WiFi and browser:

>    if [ -e /sys/class/net/ap0 ] && ([ "$reason" = "CARRIER" ] || [ "$reason" = "NOCARRIER" ] || [ "$reason" = "BOUND" ]); then

**Comment by jvonau** [comment]
Just for background info dhcpcd when starting up assigns a ipv6 address that pulls the wireless interface into a 'carrier' state then later drops the connection resulting in the 'nocarrier' state even when there are no upstream credentials for wpa_supplicant present to establish a wifi connection.
>iiab-admin@box:~ $ sudo journalctl -b 0 -l --no-pager -u dhcpcd
-- Logs begin at Wed 2020-05-13 11:37:33 EDT, end at Wed 2020-05-13 11:52:41 EDT. --
May 13 11:37:36 box.lan systemd[1]: Starting dhcpcd on all interfaces...
May 13 11:37:37 box.lan dhcpcd[426]: dev: loaded udev
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: starting wpa_supplicant
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: 50-iiab ap0 spam PREINIT
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: 50-iiab ap0 spam PREINIT
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: connected to Access Point `'
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: 50-iiab ap0 spam NOCARRIER
May 13 11:37:37 box.lan dhcpcd-run-hooks[507]: wlan0: 50-iiab ap0 spam NOCARRIER
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: 50-iiab set ap0 up NOCARRIER
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: 50-iiab ap0 spam NOCARRIER
May 13 11:37:37 box.lan dhcpcd-run-hooks[512]: wlan0: 50-iiab ap0 spam NOCARRIER
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: 50-iiab set ap0 up NOCARRIER
May 13 11:37:37 box.lan dhcpcd-run-hooks[514]: wlan0: 50-iiab set ap0 up NOCARRIER
May 13 11:37:37 box.lan dhcpcd[426]: eth0: waiting for carrier
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: waiting for carrier
May 13 11:37:37 box.lan dhcpcd[426]: br0: waiting for carrier
May 13 11:37:37 box.lan dhcpcd[426]: br0: waiting for carrier
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: carrier acquired
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: 50-iiab CARRIER change wlan0
May 13 11:37:37 box.lan dhcpcd-run-hooks[530]: wlan0: 50-iiab CARRIER change wlan0
May 13 11:37:37 box.lan dhcpcd[426]: frequency is 6 for carrier
May 13 11:37:37 box.lan dhcpcd[426]: Using 6 for carrier
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: 50-iiab set channel 6
May 13 11:37:37 box.lan dhcpcd-run-hooks[535]: wlan0: 50-iiab set channel 6
May 13 11:37:37 box.lan dhcpcd[426]: Hostapd set for 6
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: 50-iiab ap0 spam CARRIER
May 13 11:37:37 box.lan dhcpcd-run-hooks[539]: wlan0: 50-iiab ap0 spam CARRIER
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: 50-iiab set ap0 up CARRIER
May 13 11:37:37 box.lan dhcpcd-run-hooks[541]: wlan0: 50-iiab set ap0 up CARRIER
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: 50-iiab CARRIER change wlan0
May 13 11:37:37 box.lan dhcpcd-run-hooks[548]: wlan0: 50-iiab CARRIER change wlan0
May 13 11:37:37 box.lan dhcpcd[426]: frequency is 6 for carrier
May 13 11:37:37 box.lan dhcpcd[426]: Using 6 for carrier
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: 50-iiab set channel 6
May 13 11:37:37 box.lan dhcpcd-run-hooks[557]: wlan0: 50-iiab set channel 6
May 13 11:37:37 box.lan dhcpcd[426]: Hostapd set for 6
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: 50-iiab ap0 spam CARRIER
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: 50-iiab set ap0 up CARRIER
May 13 11:37:37 box.lan dhcpcd-run-hooks[570]: wlan0: 50-iiab set ap0 up CARRIER
May 13 11:37:37 box.lan dhcpcd[426]: DUID 00:01:00:01:25:d8:31:74:dc:a6:32:0d:3a:a8
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: IAID 32:0d:3a:a9
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: adding address fe80::b66a:d4b4:a4ab:4aa6
May 13 11:37:37 box.lan dhcpcd[426]: ipv6_addaddr1: Permission denied
May 13 11:37:37 box.lan dhcpcd[426]: ipv6_addaddr1: Permission denied
May 13 11:37:37 box.lan dhcpcd[426]: wlan0: carrier lost
May 13 11:37:38 box.lan dhcpcd[426]: wlan0: 50-iiab ap0 spam NOCARRIER
May 13 11:37:38 box.lan dhcpcd-run-hooks[633]: wlan0: 50-iiab ap0 spam NOCARRIER
May 13 11:37:38 box.lan dhcpcd[426]: wlan0: 50-iiab set ap0 up NOCARRIER
May 13 11:37:38 box.lan dhcpcd-run-hooks[635]: wlan0: 50-iiab set ap0 up NOCARRIER
May 13 11:37:38 box.lan dhcpcd[426]: wlan0: 50-iiab ap0 spam NOCARRIER
May 13 11:37:38 box.lan dhcpcd-run-hooks[637]: wlan0: 50-iiab ap0 spam NOCARRIER
May 13 11:37:38 box.lan dhcpcd[426]: wlan0: 50-iiab set ap0 up NOCARRIER
May 13 11:37:38 box.lan dhcpcd-run-hooks[638]: wlan0: 50-iiab set ap0 up NOCARRIER
May 13 11:37:38 box.lan dhcpcd[426]: br0: carrier acquired
May 13 11:37:38 box.lan dhcpcd[426]: br0: IAID 82:9c:db:b6
May 13 11:37:38 box.lan dhcpcd[426]: br0: adding address fe80::1e3f:280d:b642:32a4
May 13 11:37:38 box.lan dhcpcd[426]: ipv6_addaddr1: Permission denied
May 13 11:37:38 box.lan dhcpcd[426]: ipv6_addaddr1: Permission denied
May 13 11:37:38 box.lan dhcpcd[426]: br0: probing address 172.18.96.1/19
May 13 11:37:39 box.lan dhcpcd[426]: br0: soliciting an IPv6 router
May 13 11:37:42 box.lan dhcpcd[426]: eth0: carrier acquired
May 13 11:37:42 box.lan dhcpcd[426]: eth0: IAID 32:0d:3a:a8
May 13 11:37:42 box.lan dhcpcd[426]: eth0: adding address fe80::e98b:63f:c633:bcf0
May 13 11:37:42 box.lan dhcpcd[426]: ipv6_addaddr1: Permission denied
May 13 11:37:42 box.lan dhcpcd[426]: ipv6_addaddr1: Permission denied
May 13 11:37:43 box.lan dhcpcd[426]: eth0: rebinding lease of 192.168.0.181
May 13 11:37:43 box.lan dhcpcd[426]: eth0: probing address 192.168.0.181/24
May 13 11:37:43 box.lan dhcpcd[426]: eth0: soliciting an IPv6 router
May 13 11:37:43 box.lan dhcpcd[426]: br0: using static address 172.18.96.1/19
May 13 11:37:43 box.lan dhcpcd[426]: br0: adding route to 172.18.96.0/19
May 13 11:37:43 box.lan dhcpcd[426]: forked to background, child pid 710
May 13 11:37:43 box.lan systemd[1]: Started dhcpcd on all interfaces.
May 13 11:37:47 box.lan dhcpcd[710]: eth0: leased 192.168.0.181 for infinity
May 13 11:37:47 box.lan dhcpcd[710]: eth0: adding route to 192.168.0.0/24
May 13 11:37:47 box.lan dhcpcd[710]: eth0: adding default route via 192.168.0.1

**Pull request merged**